### PR TITLE
cmake: mcuboot: Add warning if missing Kconfig signing options

### LINKED ISSUE #57963: mgmt: mcumgr: image uploaded without hash wrongly shows hash of previous image
**Describe the bug**
If an application is build which supports FOTA via MCUmgr e.g. smp_svr, then an unsigned firmware is uploaded to the other slot, doing "image list" will show the same hash twice, but the second hash is invalid as the second image does not have a hash.

**Expected behavior**
Good question

**Impact**
Another good question

**Environment (please complete the following information):**
 - Commit SHA or Version used: 2ebb2bb1ab7445448d1ef561a06255313b766a94

### PR DESCRIPTION
**Pull request by nordicjm**
Adds a warning if neither Kconfig is enabled to generate an unsigned image or a signing key are set, this is not an error but warns the user in case they have forgot to set these modes or if they have been set manually and have been cleared by a cmake rebuild occurring (e.g. when bisecting).

Fixes #57963

### CODE REVIEW

**Review by de-nordic — APPROVED** [review]

**Review by tejlmand — CHANGES_REQUESTED** [review]
waiting for review from @mbolivar-nordic and / or @henrikbrixandersen

File: cmake/mcuboot.cmake
Comment (henrikbrixandersen) [reply cmake/mcuboot.cmake:27]: Thanks. This will not present an issue in our downstream workflow.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Pull request merged**
